File: Kol_2/2021_Maj/client/models/fabrika.js
export class Fabrika {
  constructor(fabrikaID, naziv, listaSilosa) {
    this.fabrikaID = fabrikaID;
    this.naziv = naziv;
    this.listaSilosa = listaSilosa;
  }

  draw(container) {
    const naziv = document.createElement("label");
    naziv.innerHTML = this.naziv;
    naziv.classList.add("nazivFabrike");

    const nazivDiv = document.createElement("div");
    nazivDiv.append(naziv);

    const silosiDiv = document.createElement("div");
    silosiDiv.classList.add("sviSilosi");

    this.listaSilosa.forEach((silos) => {
      silos.draw(silosiDiv);
    });

    container.append(nazivDiv, silosiDiv);
  }
}

File: Kol_2/2021_Maj/maj2021/Models/Fabrika.cs
using System.ComponentModel.DataAnnotations;

namespace maj2021.Models
{
    public class Fabrika
    {
        [Key]
        public int FabrikaID { get; set; }

        public required string Naziv { get; set; }

        public List<Silos>? ListaSilosa { get; set; }

        public Fabrika()
        {
            ListaSilosa = [];
        }
    }
}